src/main/java/net/snowflake/client/core/FileCacheManager.java

```diff
@@ -153,47 +152,46 @@ FileCacheManager build()
       if (Constants.getOS() == Constants.OS.WINDOWS)
       {
         this.cacheDir = new File(
-            new File(new File(new File(homeDir,
-                                       "AppData"), "Local"), "Snowflake"), "Caches");
+            new File(new File(new File(homeDir, "AppData"), "Local"), "Snowflake"), "Caches");
       }
       else if (Constants.getOS() == Constants.OS.MAC)
       {
-        this.cacheDir = new File(new File(new File(homeDir,
-                                                   "Library"), "Caches"), "Snowflake");
+        this.cacheDir = new File(new File(new File(homeDir, "Library"), "Caches"), "Snowflake");
       }
       else
       {
         this.cacheDir = new File(new File(homeDir, ".cache"), "snowflake");
       }
     }
-    if (!this.cacheDir.exists() && !this.cacheDir.mkdirs())
+
+    if (!this.cacheDir.mkdirs() && !this.cacheDir.exists())
     {
-      throw new RuntimeException(
-          String.format(
-              "Failed to locate or create the cache directory: %s", this.cacheDir)
-      );
+      LOGGER.debug("Cannot create the cache directory {}. Giving up.", this.cacheDir.getAbsolutePath());
+      return this;
     }
+    LOGGER.debug("Verified Directory {}", this.cacheDir.getAbsolutePath());
 
-    File cacheFileTmp = new File(
-        this.cacheDir, this.baseCacheFileName).getAbsoluteFile();
+    File cacheFileTmp = new File(this.cacheDir, this.baseCacheFileName).getAbsoluteFile();
     try
     {
       // create an empty file if not exists and return true.
       // If exists. the method returns false.
       // In this particular case, it doesn't matter as long as the file is
       // writable.
-      cacheFileTmp.createNewFile();
+      if (cacheFileTmp.createNewFile())
+      {
+        LOGGER.debug("Successfully created a cache file {}", cacheFileTmp);
+      }
+      else
+      {
+        LOGGER.debug("Cache file already exists {}", cacheFileTmp);
+      }
       this.cacheFile = cacheFileTmp.getCanonicalFile();
-      this.cacheLockFile = new File(
-          this.cacheFile.getParentFile(), this.baseCacheFileName + ".lck");
+      this.cacheLockFile = new File(this.cacheFile.getParentFile(), this.baseCacheFileName + ".lck");
     }
     catch (IOException | SecurityException ex)
     {
-      throw new RuntimeException(
-          String.format(
-              "Failed to touch the cache file: %s",
-              cacheFileTmp.getAbsoluteFile())
-      );
+      LOGGER.info("Failed to touch the cache file. Ignored. {}", cacheFileTmp.getAbsoluteFile());
```

Comment:
No RuntimeException.
If no cache file exists, it will continue working without the cache.

Reply:
What will be used instead of the cache?

Reply:
If the cache is usable, the JDBC driver 1) downloads the OCSP cache from ocsp.snowflakecomputing.com, 2) optionally downloads the OCSP response from the CA OCSP responder, 3) saves them into the cache file. When a new JDBC process makes a connection, it reads it from the cache first and if it doesn't contain the corresponding OCSP response, repeat from 1.

If no cache is useable, the JDBC driver 1) downloads the OCSP cache from ocsp.snowflakecomputing.com, 2) optionally downloads the OCSP response from the CA OCSP responder, 3) won't save it into the cache file. So when a new JDBC process makes a connection, since there is no saved cache, just repeat from 1.